docusaurus/docs/create-a-plugin/extend-a-plugin/add-authentication-for-data-source-plugins.md

```diff
@@ -0,0 +1,486 @@
+   const { apiKey } = secureJsonData;
+   ```
+
+:::note
+
+You can do this until the user saves the configuration; when the user saves the configuration, Grafana clears the value. After that, you can use `secureJsonFields` to determine whether the property has been configured.
+
+:::
+
+1. To securely update the secret in your plugin's configuration editor, update the `secureJsonData` object using the `onOptionsChange` prop:
+
+   ```ts
+   const onAPIKeyChange = (event: ChangeEvent<HTMLInputElement>) => {
+     onOptionsChange({
+       ...options,
+       secureJsonData: {
+         apiKey: event.target.value,
+       },
+     });
+   };
+   ```
+
+1. Define a component that can accept user input:
+
+   ```ts
+   <Input
+     type="password"
+     placeholder={secureJsonFields?.apiKey ? 'configured' : ''}
+     value={secureJsonData.apiKey ?? ''}
+     onChange={onAPIKeyChange}
+   />
+   ```
+
+1. Optional: If you want the user to be able to reset the API key, then you need to set the property to `false` in the `secureJsonFields` object:
+
+   ```ts
+   const onResetAPIKey = () => {
+     onOptionsChange({
+       ...options,
+       secureJsonFields: {
+         ...options.secureJsonFields,
+         apiKey: false,
+       },
+       secureJsonData: {
+         ...options.secureJsonData,
+         apiKey: '',
+       },
+     });
+   };
+   ```
+
+Now that users can configure secrets, the next step is to see how we can add them to our requests.
+
+## Authenticate using the data source proxy
+
+Once the user has saved the configuration for a data source, the secret data source configuration will no longer be available in the browser. Encrypted secrets can only be accessed on the server. So how do you add them to your request?
+
+The Grafana server comes with a proxy that lets you define templates for your requests: _proxy routes_. Grafana sends the proxy route to the server, decrypts the secrets along with other configuration, and adds them to the request before sending it.
+
+:::note
+
+Be sure not to confuse the data source proxy with the [auth proxy](https://grafana.com/docs/grafana/latest/setup-grafana/configure-security/configure-authentication/auth-proxy/index.md). The data source proxy is used to authenticate a data source, while the auth proxy is used to log into Grafana itself.
+
+:::
+
+### Add a proxy route to your plugin
+
+To forward requests through the Grafana proxy, you need to configure one or more _proxy routes_. A proxy route is a template for any outgoing request that is handled by the proxy. You can configure proxy routes in the [plugin.json](https://grafana.com/docs/grafana/latest/developers/plugins/metadata/) file.
+
+1. Add the route to `plugin.json`:
+
+   ```json
+   "routes": [
+     {
+       "path": "example",
+       "url": "https://api.example.com"
+     }
+   ]
+   ```
+
+    :::note
+
+    You need to restart the Grafana server every time you make a change to your `plugin.json` file.
+
+    :::
```

Comment:
```suggestion
  :::note

  You need to restart the Grafana server every time you make a change to your `plugin.json` file.

  :::
```
Need to fix the indentation for this to display properly
<img width="930" alt="Screenshot 2023-08-16 at 17 37 30" src="https://github.com/grafana/plugin-tools/assets/19860021/52559bf1-2ae2-4858-9ba2-4bbbc0c08dd5">